integration-test/src/main/java/com/sequenceiq/it/cloudbreak/CredentialModifyClusterTests.java

```diff
@@ -0,0 +1,259 @@
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName), "a credential is created.");
+        given(cloudProvider.aValidStackCreated()
+                .withName(clusterName), "a stack is created");
+        given(StackOperation.request());
+        when(StackOperation.stop());
+        when(Stack.get());
+        then(Stack.waitAndCheckClusterAndStackStoppedStatus(), "stack has been stopped");
+    }
+
+    @Test(dataProvider = "providername", priority = 40)
+    public void testStartCluster(CloudProvider cloudProvider, String clusterName, String credentialName) throws Exception {
+        given(CloudbreakClient.created());
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName), "a credential is created.");
+        given(cloudProvider.aValidStackCreated()
+                .withName(clusterName), "a stack is created");
+        given(StackOperation.request());
+        when(StackOperation.start());
+        when(Stack.get());
+        then(Stack.waitAndCheckClusterAndStackAvailabilityStatus(), "stack has been started");
+        then(Stack.checkClusterHasAmbariRunning(
+                getTestParameter().get(CloudProviderHelper.DEFAULT_AMBARI_PORT),
+                getTestParameter().get(CloudProviderHelper.DEFAULT_AMBARI_USER),
+                getTestParameter().get(CloudProviderHelper.DEFAULT_AMBARI_PASSWORD)),
+                "ambari check");
+    }
+
+    @Test(alwaysRun = true, dataProvider = "providername", priority = 50)
+    public void testTerminateCluster(CloudProvider cloudProvider, String clusterName, String credentialName) throws Exception {
+        given(CloudbreakClient.created());
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName), "a credential is created.");
+        given(cloudProvider.aValidStackCreated()
+                .withName(clusterName), "a stack is created");
+        when(Stack.delete());
+        then(Stack.waitAndCheckClusterDeleted(), "stack has been deleted");
+    }
+
+    @DataProvider(name = "providernameblueprintimage")
+    public Object[][] providerAndImage() throws Exception {
+        String blueprint = getTestParameter().get("blueprintName");
+        String provider = getTestParameter().get("provider").toLowerCase();
+        String imageDescription = getTestParameter().get("image");
+
+        CloudProvider cloudProvider = CloudProviderHelper.providerFactory(provider, getTestParameter());
+
+        String clusterName = getTestParameter().get("clusterName");
+        String credentialName = getTestParameter().get("credentialName");
+        String image = getImageId(provider, imageDescription, blueprint);
+        return new Object[][]{
+                {cloudProvider, clusterName, credentialName, blueprint, image}
+        };
+    }
+
+    @DataProvider(name = "providernamehostgroupdesiredno")
+    public Object[][] providerAndHostgroup() throws Exception {
```

Comment:
unnecessary exception declaration in method signature. there is no checked exception coming from inside.